linkerd/core/src/main/scala/io/buoyant/linkerd/ActiveTracer.scala

```diff
@@ -0,0 +1,170 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util.{Future, Stopwatch}
+import io.buoyant.linkerd.ActiveTracer.formatDelegation
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.router.RoutingFactory.BaseDtab
+
+private[linkerd] case class RouterContext(
+  routerLabel: String,
+  elapsed: Stopwatch.Elapsed,
+  serviceName: String,
+  clientName: String,
+  selectedAddresses: Option[Set[String]],
+  selectedAddress: String,
+  dtabResolution: List[String]
+) {
+  def formatRouterContext: String = {
+    s"""|--- Router: $routerLabel ---
+          |request duration: ${elapsed().inMillis.toString} ms
+          |service name: $serviceName
+          |client name: $clientName
+          |addresses: [${selectedAddresses.getOrElse(Set.empty).mkString(", ")}]
+          |selected address: $selectedAddress
+          |dtab resolution:
+          |${dtabResolution.map("  " + _).mkString(System.lineSeparator)}
+          |""".stripMargin
+  }
+}
+
+private[linkerd] object RouterContextBuilder {
```

Comment:
RouterContextFormatter perhaps?  Builder suggests the builder pattern to me.

Reply:
Yea, I can see that. I realize I can also see that I could move the contents of the `apply` method from RouterContextBuilder to `ActiveTracer`